library/asn1parse.c

```diff
@@ -280,6 +285,10 @@ int mbedtls_asn1_traverse_sequence_of(
     while (*p < end) {
         unsigned char const tag = *(*p)++;
 
+        if (mbedtls_asn1_bad_tag(tag)) {
```

Comment:
I don't understand why we'd specifically reject 0 here, but accept all other tags. The caller will need to filter tags anyway.

I'm not sure why `mbedtls_asn1_traverse_sequence_of` is a public function anyway. (I approved it in https://github.com/ARMmbed/mbed-crypto/pull/263, but now I don't know why. Maybe because it's unit-tested and at the time we couldn't unit-test private functions? But unit tests for `mbedtls_asn1_get_sequence_of` would have sufficed. Or maybe because it was needed for future X.509 work that we never finished?) Inside the library, it's only used by `mbedtls_asn1_get_sequence_of` which doesn't need all this flexibility, in particular `tag` will be constrained to one specific value. So I don't think we need to make this function more robust.